src/Workspaces/Core/Portable/Utilities/TaskExtensions.cs

```diff
@@ -156,6 +162,8 @@ public static T WaitAndGetResult_CanCallOnBackground<T>(this Task<T> task, Cance
             // We do not want this, so we pass the LazyCancellation flag to the TPL which implements
             // the behavior we want.
 
+            Contract.ThrowIfNull(continuationFunction, nameof(continuationFunction));
+
             Func<Task, TResult> outerFunction = t =>
             {
                 try
```

Comment:
:memo: The second bug I mentioned is a Watson report indicating a crash within the `FatalError.ReportUnlessCanceled(e)` right below this. The original exception type is not revealed in the reports I saw, but based on the stack it's likely that the original exception is a `NullReferenceException` due to `continuationFunction` being null.